tests/pathtourl.cpp

```diff
@@ -0,0 +1,96 @@
+
+#include <cstdlib>
+#include <iostream>
+
+#include "ada.h"
+
+
+/**
+ * Node.js has a function called url.pathToFileURL(path).
+ * https://nodejs.org/api/url.html#urlpathtofileurlpath
+ *
+ * It is explained as follows...
+ *
+ * This function ensures that path is resolved absolutely,
+ * and that the URL control characters are correctly encoded when converting into a File URL.
+ *
+ * new URL('/foo#1', 'file:');           // Incorrect: file:///foo#1
+ * pathToFileURL('/foo#1');              // Correct:   file:///foo%231 (POSIX)
+ *
+ * new URL('/some/path%.c', 'file:');    // Incorrect: file:///some/path%.c
```

Comment:
Safari returns `"file:///some/path%.c"` for href for this value.

Reply:
Yes. Firefox too...
I get `file:///some/path%.c`. It does not try to percent encode the '%' character.